python/mlc_chat/compiler/quantization/no_quantization.py

```diff
@@ -0,0 +1,99 @@
+"""The no quantization config"""
+import logging
+from dataclasses import dataclass
+from typing import Any, List
+
+from tvm.relax.frontend import nn
+from tvm.runtime import NDArray
+
+from ..loader import QuantizeMapping
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class NoQuantize:  # pylint: disable=too-many-instance-attributes
+    """Configuration for no quantization"""
+
+    name: str
+    kind: str
+    model_dtype: str  # "float16", "float32"
+
+    def __post_init__(self):
+        assert self.kind == "no-quant"
+
+    def quantize_model(
+        self,
+        model: nn.Module,
+        quant_map: QuantizeMapping,
+        name_prefix: str,
+    ) -> nn.Module:
+        """
+        Quantize model without quantization
+
+        Parameters
+        ----------
+        model : nn.Module
+            The non-quantized nn.Module.
+
+        quant_map : QuantizeMapping
+            The quantize mapping with name mapping and func mapping.
+
+        name_prefix : str
+            The name prefix for visited weight.
+
+        Returns
+        -------
+        ret : nn.Module
+            The original nn.Module.
+        """
+
+        class _Mutator(nn.Mutator):
+            def __init__(self, config: NoQuantize, quant_map: QuantizeMapping) -> None:
+                super().__init__()
+                self.config = config
+                self.quant_map = quant_map
+
+            def visit_module(self, name: str, node: nn.Module) -> Any:
+                """
+                The visiting method for quantization of nn.Module nodes.
+
+                Parameters
+                ----------
+                name : str
+                    The name of the current node.
+
+                node : nn.Module
+                    The current node of nn.Module to mutate.
+
+                Returns
+                ------
+                ret_node: Any
+                    The new node to replace current node.
+                """
+                if isinstance(node, (nn.Linear, nn.MultiLinear, nn.Embedding)):
+                    weight_name = f"{name}.weight"
+                    self.quant_map.param_map[weight_name] = [weight_name]
+                    self.quant_map.map_func[weight_name] = self.config.quantize_weight
+                return self.visit(name, node)
+
+        model.to(dtype=self.model_dtype)
+        mutator = _Mutator(self, quant_map)
```

Comment:
Do we even need a mutator in `NoQuantize`? I suppose `model.to` has already converted the model to `model_dtype`

Reply:
yes, removed.